lib/Controller/SocialApiController.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * @copyright Copyright (c) 2020 Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @author Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Contacts\Controller;
+
+use OCP\AppFramework\ApiController;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Http\TemplateResponse;
+// use OCP\IInitialStateService;
+use OCP\IConfig;
+use OCP\Contacts\IManager;
+use OCP\L10N\IFactory;
+use OCP\IRequest;
+
+
+class SocialApiController extends ApiController {
+
+	protected $appName;
+
+	// /** @var IInitialStateService */
+	// private $initialStateService;
+
+	/** @var IFactory */
+	private $languageFactory;
+	/** @var IManager */
+	private  $manager;
+	/** @var IConfig */
+	private  $config;
+
+	public function __construct(string $AppName,
+								IRequest $request,
+								IManager $manager,
+								IConfig $config,
+								// IInitialStateService $initialStateService,
+								IFactory $languageFactory) {
+		parent::__construct($AppName, $request);
+
+		$this->appName = $AppName;
+		// $this->initialStateService = $initialStateService;
+		$this->languageFactory = $languageFactory;
+		$this->manager = $manager;
+		$this->config = $config;
+	}
+
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * generate download url for a social entry (based on type of data requested)
+	 */
+	protected function getSocialConnector(array $socialentry, string $type) : ?string {
```

Comment:
Missing docs :)

Reply:
Had this in the beginning and missunderstood your comment, will put it back in!